cruise-control-core/src/main/java/com/linkedin/cruisecontrol/metricdef/MetricDef.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.cruisecontrol.metricdef;
+
+import com.linkedin.cruisecontrol.CruiseControlUtils;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import com.linkedin.cruisecontrol.monitor.sampling.aggregator.MetricSampleAggregator;
+
+
+/**
+ * The definition of metrics. Each metric will be assigned a metric id and used by the {@link MetricSampleAggregator}
+ * to look up the metrics.
+ */
+public class MetricDef {
+  private final AtomicInteger _nextIndex;
+  private final Map<String, MetricInfo> _metricNameToIndex;
+  private final SortedMap<Integer, MetricInfo> _indexToMetricInfo;
+  private final Set<Integer> _metricsToPredict;
+
+  public MetricDef() {
+    _nextIndex = new AtomicInteger(0);
+    _metricNameToIndex = new HashMap<>();
+    _indexToMetricInfo = new TreeMap<>();
```

Comment:
Since metrics are ordered sequentially and are number this can just be an ArrayList.